nrf_rpc/doc/protocol_specification.rst

```diff
@@ -0,0 +1,126 @@
+.. _nrf_rpc_protocol_specification:
+
+nRF RPC protocol specification
+##############################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Two processors that communicate with each other using the nRF RPC library follow the nRF RPC protocol.
+The nRF RPC protocol specifies the binary format and rules for constructing packets that are exchanged within an nRF RPC communication.
+
+The nRF RPC packets that are constructed by the nRF RPC core library are relayed to the selected transport layer, where they can be additionally encoded to assure a reliable delivery of a packet to the other processor using the selected medium.
+The nRF RPC transports specification is outside the scope of this document.
+
+The types of communication, threading model and other general concepts of the nRF RPC are described in :ref:`nrf_rpc_architecture`.
+
+nRF RPC packet format
+*********************
+
+Each nRF RPC packet consists of a 5-byte header and optional, variable-length payload:
+
+.. table::
+   :align: center
+
+   +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   |0                              |1                              |
+   +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   |0  |1  |2  |3  |4  |5  |6  |7  |0  |1  |2  |3  |4  |5  |6  |7  |
+   +===+===+===+===+===+===+===+===+===+===+===+===+===+===+===+===+
+   | Type [\| Source Context ID]   | Command ID                    |
+   +-------------------------------+-------------------------------+
+   | Destination Context ID        | Source Group ID               |
+   +-------------------------------+-------------------------------+
+   | Desination Group ID           | [Payload...]                  |
+   +-------------------------------+-------------------------------+
+   |                             [...]                             |
+   +---------------------------------------------------------------+
+
+A description of the various fields and their meaning:
+
+``Type [| Source Context ID]``: 8 bits
+  The packet type determines the function of the packet and it can be one of the following values:
+
+  * **“0x00”** - event
+  * **“0x01”** - response
+  * **“0x02”** - event acknowledgement
+  * **“0x03”** - error report
+  * **“0x04”** - initialization packet
+  * **“0x80”** - command
```

Comment:
The formatting for these seems odd. Since they're values, shouldn't they be ``0x00``, and so on?

Reply:
Right, I copied this convention from another specification, but `0x00` will be more consistent with Zephyr docs.